datahub-web-react/src/app/ingestV2/ManageIngestionPage.tsx

```diff
@@ -82,22 +82,47 @@ export const ManageIngestionPage = () => {
     const isShowNavBarRedesign = useShowNavBarRedesign();
     const [showCreateSourceModal, setShowCreateSourceModal] = useState<boolean>(false);
     const [showCreateSecretModal, setShowCreateSecretModal] = useState<boolean>(false);
-    const [hideSystemSources, setHideSystemSources] = useState(true);
-
-    const {
-        isLoading: isCapabilitySummaryLoading,
-        error: isCapabilitySummaryError,
-        isProfilingSupported,
-    } = useCapabilitySummary();
     const history = useHistory();
+    const location = useLocation();
     const shouldPreserveParams = useRef(false);
 
-    if (!isCapabilitySummaryLoading && !isCapabilitySummaryError) {
-        console.log(
-            'Example to be removed when is actually used for something is profiling support for bigquery',
-            isProfilingSupported('bigquery'),
-        );
-    }
+    // Parse URL parameters for filters
+    const params = QueryString.parse(location.search, { arrayFormat: 'comma' });
+    const hideSystemFromUrl = params.hideSystem === 'true';
+    const sourceFilterFromUrl = params.sourceFilter ? Number(params.sourceFilter) : undefined;
+
+    const [hideSystemSources, setHideSystemSources] = useState(hideSystemFromUrl ?? true);
+
+    // Update URL when filters change
+    const updateUrlWithFilters = useCallback(
+        (newHideSystem?: boolean, newSourceFilter?: number) => {
+            const currentParams = QueryString.parse(location.search, { arrayFormat: 'comma' });
+            const updatedParams = { ...currentParams };
+
+            if (newHideSystem !== undefined) {
+                updatedParams.hideSystem = newHideSystem.toString();
+            }
+            if (newSourceFilter !== undefined) {
+                updatedParams.sourceFilter = newSourceFilter.toString();
+            }
+
+            const newSearch = QueryString.stringify(updatedParams, { arrayFormat: 'comma' });
```

Comment:
nit: i think we can reuse `useUrlAueryParams` here to make your life easier and reuse that logic - it has a `setValue` method returned that updates the query param in the url too